docs/website/content/v0.3/en/guides/cloud/aws.md

```diff
@@ -2,18 +2,225 @@
 title: 'AWS'
 ---
 
-First, create the AMI:
+## Creating a Cluster via the AWS CLI
+
+In this guide we will create an HA Kubernetes cluster with 3 worker nodes.
+We assume an existing VPC.
+
+### Create the Subnet
+
+```bash
+aws ec2 create-subnet \
+    --region ${REGION} \
+    --vpc-id ${VPC} \
+    --cidr-block 10.0.1.0/24
+```
+
+### Create the AMI
+
+#### Prepare the Import Prerequisites
+
+##### Create the S3 Bucket
+
+```bash
+aws s3api create-bucket \
+    --bucket talos-aws-tutorial-${ACCOUNT} \
+    --create-bucket-configuration LocationConstraint=${REGION} \
+    --acl private
+```
+
+##### Create the `vmimport` Role
+
+In order to create an AMI, ensure that the `vmimport` role exists as described in the [official AWS documentation](https://docs.aws.amazon.com/vm-import/latest/userguide/vmie_prereqs.html#vmimport-role).
+
+##### Create the Image Snapshot
+
+```bash
+curl -LO https://github.com/talos-systems/talos/releases/download/v0.3.0-alpha.2/aws.tar.gz
+tar -xvf aws.tar.gz
+aws s3 cp aws.raw s3://${BUCKET}/talos-aws-tutorial.raw
+aws ec2 import-snapshot \
+    --region ${REGION} \
+    --description "Talos kubernetes tutorial" \
+    --disk-container "Format=raw,UserBucket={S3Bucket=${BUCKET},S3Key=talos-aws-tutorial.raw}"
+```
+
+Save the `SnapshotId`, as we will need it once the import is done.
+To check on the status of the import, run:
 
 ```bash
-docker run \
-    --rm \
-    --volume $HOME/.aws/credentials:/root/.aws/credentials \
-    --env AWS_DEFAULT_PROFILE=${PROFILE} \
-    --env AWS_DEFAULT_REGION=${REGION} \
-    talos-systems/talos:latest ami -var regions=${COMMA_SEPARATED_LIST_OF_REGIONS}
+aws ec2 describe-import-snapshot-tasks \
+    --region ${REGION} \
+    --import-task-ids
 ```
 
-Once the AMI is created, you can now start an EC2 instance using the AMI ID.
-Provide the proper configuration as the instance's user data.
+Once the `SnapshotTaskDetail.Status` indicates `completed`, we can register the image.
+
+##### Register the Image
+
+```bash
+aws ec2 register-image \
+    --region ${REGION} \
+    --block-device-mappings "DeviceName=/dev/xvda,VirtualName=talos,Ebs={DeleteOnTermination=true,SnapshotId=${SNAPSHOT},VolumeSize=4,VolumeType=gp2}" \
+    --root-device-name /dev/xvda \
+    --virtualization-type hvm \
+    --architecture x86_64 \
+    --ena-support \
+    --name talos-aws-tutorial-ami
+```
+
+We now have an AMI we can use to create our cluster.
```

Comment:
save AMI id in `$AMI` ?

Reply:
Yeah I'm realizing that I need to explain when to save what.